cmd/snap-failure/cmd_snapd.go

```diff
@@ -133,6 +133,10 @@ func (c *cmdSnapd) Execute(args []string) error {
 	}
 
 	logger.Noticef("restarting snapd socket")
+	// we need to reset the failure state to be able to restart again
+	if output, err := exec.Command("systemctl", "reset-failed", "snapd.socket").CombinedOutput(); err != nil {
+		return osutil.OutputErr(output, err)
+	}
```

Comment:
Thanks for this, I think we should try to be a bit more robust in this eventually and try to restart snapd.socket anyways even if we can't reset-failed, but this is fine for now.